pkg/server/handler/auth.go

```diff
@@ -712,3 +712,162 @@ func (h *PasswordHandler) Handle(payload *router.Payload, response *router.Respo
 	}.WithRouterPayload(payload))
 	h.PwHousekeeper.Housekeep(info.ID)
 }
+
+type resetPasswordPayload struct {
+	AuthID          string `mapstructure:"auth_id"`
+	NewPassword     string `mapstructure:"password"`
+	Invalidate      bool   `mapstructure:"invalidate"`
+	passwordChecker *audit.PasswordChecker
+	conn            skydb.Conn
+	db              skydb.Database
+	userRecord      *skydb.Record
+	authInfo        *skydb.AuthInfo
+}
+
+func (payload *resetPasswordPayload) Decode(data map[string]interface{}) skyerr.Error {
+	if err := mapstructure.Decode(data, payload); err != nil {
+		return skyerr.NewError(skyerr.BadRequest, "fails to decode the request payload")
+	}
+	if err := payload.inject(); err != nil {
+		return err
+	}
+	return payload.Validate()
+}
+
+func (payload *resetPasswordPayload) inject() skyerr.Error {
+	// We cannot use preprocessor to inject user
+	// because we accepts AuthID in the payload
+	// instead of from access token header.
+	// We assume ResetPasswordHandler requires master key
+	// so that this injection is safe
+	payload.authInfo = &skydb.AuthInfo{}
+	payload.userRecord = &skydb.Record{}
+	if err := payload.conn.GetAuth(payload.AuthID, payload.authInfo); err != nil {
+		return skyerr.NewError(skyerr.UnexpectedAuthInfoNotFound, err.Error())
+	}
+	if err := payload.db.Get(skydb.NewRecordID("user", payload.AuthID), payload.userRecord); err != nil {
+		return skyerr.NewError(skyerr.UnexpectedUserNotFound, err.Error())
+	}
+	return nil
+}
+
+func (payload *resetPasswordPayload) Validate() skyerr.Error {
+	var userData map[string]interface{}
+	if payload.userRecord != nil {
+		userData = map[string]interface{}(payload.userRecord.Data)
+	}
+	return payload.passwordChecker.ValidatePassword(audit.ValidatePasswordPayload{
+		AuthID:        payload.AuthID,
+		PlainPassword: payload.NewPassword,
+		UserData:      userData,
+		Conn:          payload.conn,
+	})
```

Comment:
I think for the same reason, the passwordChecker call should also in `Handle` func.

Let change the changePasswordHandler too.